diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8e42ebd..4715618 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -123,12 +123,12 @@ ELSE ( NOT ZLIB_FOUND)
 ENDIF( NOT ZLIB_FOUND)
 
 #SET(LibSolv_USE_STATIC_LIBS ON)
-FIND_PACKAGE(LibSolv REQUIRED ext)
-IF ( NOT LibSolv_FOUND )
-  MESSAGE( FATAL_ERROR " libsolv not found" )
-ELSE()
+#jca FIND_PACKAGE(LibSolv REQUIRED ext)
+#jca IF ( NOT LibSolv_FOUND )
+  #jca MESSAGE( FATAL_ERROR " libsolv not found" )
+#jca ELSE()
   INCLUDE_DIRECTORIES( ${LibSolv_INCLUDE_DIRS} )
-ENDIF()
+#jca ENDIF()
 
 # libsolv uses expat and has open references to it:
 FIND_PACKAGE(EXPAT REQUIRED)
diff --git a/cmake/modules/FindRpm.cmake b/cmake/modules/FindRpm.cmake
index 29696e3..0a2a14e 100644
--- a/cmake/modules/FindRpm.cmake
+++ b/cmake/modules/FindRpm.cmake
@@ -5,11 +5,16 @@ if(RPM_INCLUDE_DIR AND RPM_LIBRARY)
 endif(RPM_INCLUDE_DIR AND RPM_LIBRARY)
 
 set(RPM_LIBRARY)
+set(RPMIO_LIBRARY)
+set(LZMA_LIBRARY)
+set(PTHREAD_LIBRARY)
+set(POPT_LIBRARY)
 set(RPM_INCLUDE_DIR)
 
 FIND_PATH(RPM_INCLUDE_DIR rpm/rpmdb.h
 	/usr/include
 	/usr/local/include
+	/opt/profbuild/include
 )
 
 set(RPM_SUSPECT_VERSION "RPM_SUSPECT_VERSION-NOTFOUND" )
@@ -38,6 +43,28 @@ FIND_LIBRARY(RPM_LIBRARY NAMES rpm
 	PATHS
 	/usr/lib
 	/usr/local/lib
+	/opt/profbuild/lib
+)
+FIND_LIBRARY(RPMIO_LIBRARY NAMES rpmio
+	PATHS
+	/usr/lib
+	/usr/local/lib
+	/opt/profbuild/lib
+)
+FIND_LIBRARY(PTHREAD_LIBRARY NAMES pthread
+	PATHS
+	/usr/lib
+	/usr/local/lib
+)
+FIND_LIBRARY(LZMA_LIBRARY NAMES lzma
+	PATHS
+	/usr/lib
+	/usr/local/lib
+)
+FIND_LIBRARY(POPT_LIBRARY NAMES popt
+	PATHS
+	/usr/lib
+	/usr/local/lib
 )
 
 if(RPM_INCLUDE_DIR AND RPM_LIBRARY)
@@ -47,4 +74,4 @@ else(RPM_INCLUDE_DIR AND RPM_LIBRARY)
    MESSAGE( STATUS "rpm not found")
 endif(RPM_INCLUDE_DIR AND RPM_LIBRARY)
 
-MARK_AS_ADVANCED(RPM_INCLUDE_DIR RPM_LIBRARY)
\ No newline at end of file
+MARK_AS_ADVANCED(RPM_INCLUDE_DIR RPM_LIBRARY RPMIO_LIBRARY LZMA_LIBRARY POPT_LIBRARY PTHREAD_LIBRARY)
diff --git a/zypp/CMakeLists.txt b/zypp/CMakeLists.txt
index db4d53f..039ab52 100644
--- a/zypp/CMakeLists.txt
+++ b/zypp/CMakeLists.txt
@@ -922,7 +922,7 @@ ADD_DEPENDENCIES(zypp schema_header)
 # System libraries
 SET(UTIL_LIBRARY util)
 TARGET_LINK_LIBRARIES(zypp ${UTIL_LIBRARY} )
-TARGET_LINK_LIBRARIES(zypp ${RPM_LIBRARY} )
+TARGET_LINK_LIBRARIES(zypp ${RPM_LIBRARY} ${RPMIO_LIBRARY} ${LZMA_LIBRARY} ${POPT_LIBRARY} ${PTHREAD_LIBRARY} )
 TARGET_LINK_LIBRARIES(zypp ${GETTEXT_LIBRARIES} )
 TARGET_LINK_LIBRARIES(zypp ${CURL_LIBRARY} )
 TARGET_LINK_LIBRARIES(zypp ${LIBXML_LIBRARY} )
